src/EventEdition/LocationPanel.vala

```diff
@@ -187,6 +189,46 @@ public class Maya.View.EventEdition.LocationPanel : Gtk.Grid {
             debug (error.message);
         }
     }
+
+    private async void find_location (double latitude, double longitude) {
+        if (find_cancellable != null)
+            find_cancellable.cancel ();
```

Comment:
As far as I can see, find_cancellable is never created?  It is recommended to create a new cancellable when required rather than re-using the old one.